### PR TITLE
[full-ci] [tests-only] Added tests for PROPFIND as sharee

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
As sharee, PROPFIND the shares, when two users share resources with the same name using graph API.

Added Scenario:
- `Scenario: sharee PROPFIND a shares when multiple user shares resources with same name`

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/8493

## How Has This Been Tested?

- Locally
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 1]
```diff
@@ -3646,26 +3648,47 @@ public function userSendsPropfindRequestToSpaceWithHeaders(string $user, string
...
+		];
 		return  WebDavHelper::propfind(
 			$this->featureContext->getBaseUrl(),
 			$this->featureContext->getActualUsername($user),
 			$this->featureContext->getPasswordForUser($user),
 			$resource,
 			$properties,
 			$this->featureContext->getStepLineRef(),
-			"0",
+			$folderDepth,
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3688]: we can use default ocis value which is 1. then might not have to create new step
```suggestion
			"1",
```

Comment (grgprarup) [reply tests/acceptance/features/bootstrap/SpacesContext.php:3688]: The step can be reusable for multiple depths.

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/SpacesContext.php:3688]: True, but we don't know when we might need it. But fine with me :+1:

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -3646,26 +3648,47 @@ public function userSendsPropfindRequestToSpaceWithHeaders(string $user, string
...
 	 * @param array|null $headers
+	 * @param string|null $folderDepth
 	 *
 	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 *
 	 * @throws JsonException
 	 */
-	public function sendPropfindRequestToSpace(string $user, string $spaceName, ?string $resource = "", ?array $headers = []): ResponseInterface {
+	public function sendPropfindRequestToSpace(string $user, string $spaceName, ?string $resource = "", ?array $headers = [], ?string $folderDepth = "0"): ResponseInterface {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3658]: `1` should be default value
```suggestion
	public function sendPropfindRequestToSpace(string $user, string $spaceName, ?string $resource = "", ?array $headers = [], ?string $folderDepth = "1"): ResponseInterface {
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 3]
```diff
@@ -3596,22 +3596,41 @@ public function searchResultShouldContainSpace(string $user, string $spaceName):
 	}
 
 	/**
-	 * @When /^user "([^"]*)" sends PROPFIND request to space "([^"]*)" using the WebDAV API$/
-	 * @When /^user "([^"]*)" sends PROPFIND request from the space "([^"]*)" to the resource "([^"]*)" using the WebDAV API$/
+	 * @When /^user "([^"]*)" sends PROPFIND request to space "([^"]*)" with depth "([^"]*)" using the WebDAV API$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3599]: we can use this step for default depth:
`When user "Brian" sends PROPFIND request from the space "Shares" using the WebDAV API`
```suggestion
         * @When /^user "([^"]*)" sends PROPFIND request to space "([^"]*)" using the WebDAV API$/
	 * @When /^user "([^"]*)" sends PROPFIND request to space "([^"]*)" with depth "([^"]*)" using the WebDAV API$/
```

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/SpacesContext.php:3599]: I see the step below

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/SpacesContext.php:3599]: Can they be combined? or could put these together
```php
@When /^user "([^"]*)" sends PROPFIND request to space "([^"]*)" using the WebDAV API$/
@When /^user "([^"]*)" sends PROPFIND request to space "([^"]*)" with depth "([^"]*)" using the WebDAV API$/
```

File: tests/acceptance/features/apiSharingNg/propfindShares.feature [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | Viewer       |
+    And user "Carol" has sent the following share invitation:
+      | resource        | textfile.txt |
+      | space           | Personal     |
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | Viewer       |
+    When user "Brian" sends PROPFIND request from the space "Shares" to the resource "/" with depth "1" using the WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/propfindShares.feature:29]:
```suggestion
    When user "Brian" sends PROPFIND request to space "Shares" using the WebDAV API
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**